packages/vue-query/src/useMutation.ts

```diff
@@ -125,11 +137,10 @@ export function useMutation<
     })
   }
 
-  watch(
-    [() => arg1, () => arg2, () => arg3],
+  watch(options,
```

Comment:
Could you please explain the reason for this change?
Also could you add a test for it to verify if deep property watching still works?

Reply:
Essentially just to reuse the logic for providing an argument to queryClient.defaultMutationOptions and the initial declaration for the observer var. Realized 143 is definitely incorrect however, the argument should just be options.value!

Also reduces the redundancy in defining when the mutation options are recalced, but not super important. lmk if I should just do something more similar to what was already implemented :)

Will absolutely write a test. thanks for being patient with this really appreciate it

Reply:
In terms of testing deep reactivity - a library I'm trying to replicate (the reason I'm making this PR) is passing an object as the mutationKey so I think this should be sufficient basis for testing deep reactivity

Reply:
Since you have removed `defaultMutationOptions` from watch, i think you need to move it to `computed` and remove it as well from line 124.

As for deep reactivity, i'm not sure if `computed` can detect such changes. And since now watcher is based on `computed` result it will not run until `computed` value is updated. You are removing reacivity from all options when calling `parseMutationArgs` there, so this is why i'm afraid that it will only work when you update the immediate child prop of an object for example.
I might be wrong though, that is why it would be good to write a test case that is checking this, and will make sure that it would not break in the future.

Reply:
test case for deep reactivity is working - let me know if it is bad style thanks

Reply:
"Since you have removed defaultMutationOptions from watch, i think you need to move it to computed and remove it as well from line 124." 

are you suggested to make queryClient.defaultMutationOptions(options.value) its own computed property? Definitely think that can be done, just want to clarify

Reply:
Was thinking of
```ts
const options = computed(() => {
  return queryClient.defaultMutationOptions(parseMutationArgs(arg1, arg2, arg3))
})
```
and reuse it in both places, since you are already moving stuff

Reply:
in the instantiation of the queryClient we need options.value.queryClientKey, so above snippet is not achievable as far as I can see...

Reply:
As far as I can see that wouldn't be possible because in order to create the const queryClient we need the options.value.queryClient and useQueryClient(options.value.queryClientKey) as a fallback